### DISCUSSION TITLE
Modify Session Object Value ?

### DISCUSSION THREAD

**Discussion by sushilbalami** [nextauthjs/next-auth#2163] [Help]
Can I modify the session object?
I am using credentials provider so.., I am authenticating with a custom database so
although I put this obj
```
"role": "user",
"_id": "60c1f6d9abaa0e2670b06b85",
"email": "sush@gmail.com",
"createdAt": "2021-06-10T11:26:17.308Z",
"updatedAt": "2021-06-10T11:26:17.308Z",
"__v": 0 
```

this returns

```
{
  user: { name: null, email: 'sush@gmail.com', image: null },
  expires: '2021-07-11T07:08:46.365Z'
}
```

on getSession({req});

so any way to add custom key-value or custom value on session obj?

**Comment by balazsorban44** [comment]
https://next-auth.js.org/configuration/callbacks#session-callback

**Reply by sushilbalami** [reply]
only name, email and image keys are parsed why so ?
`user: { name: 'sushilbalami', email: '***@gmail.com', image: null },
`

**Reply by sushilbalami** [reply]
i tried putting
`
session.user.id = customId,
`
but only name, email, image values are only parsed
